platform-tests/src/test/java/org/junit/platform/console/tasks/ConsoleTestExecutorTests.java

```diff
@@ -120,11 +120,31 @@ void hasStatusCode1ForFailingContainer() throws Exception {
 		dummyTestEngine.addContainer("failingContainer", FAILING_BLOCK);
 
 		ConsoleTestExecutor task = new ConsoleTestExecutor(options, () -> createLauncher(dummyTestEngine));
-		int exitCode = ConsoleLauncherExecutionResult.computeExitCode(task.execute(dummyWriter()));
+		int exitCode = ConsoleLauncherExecutionResult.computeExitCode(task.execute(dummyWriter()), options);
 
 		assertThat(exitCode).isEqualTo(1);
 	}
 
+	@Test
+	void hasStatusCodeMinus1ForNoTestsAndHasOptionFailIfNoTestsFound() throws Exception {
+
+		options.setFailIfNoTests(true);
+
+		ConsoleTestExecutor task = new ConsoleTestExecutor(options, () -> createLauncher(dummyTestEngine));
+		int exitCode = ConsoleLauncherExecutionResult.computeExitCode(task.execute(dummyWriter()), options);
+
+		assertThat(exitCode).isEqualTo(-1);
+	}
+
+	@Test
+	void hasStatusCode0ForNoTestsAndNotFailIfNoTestsFound() throws Exception {
+
+		ConsoleTestExecutor task = new ConsoleTestExecutor(options, () -> createLauncher(dummyTestEngine));
+		int exitCode = ConsoleLauncherExecutionResult.computeExitCode(task.execute(dummyWriter()), options);
+
+		assertThat(exitCode).isEqualTo(0);
+	}
+
```

Comment:
I think we make all of these tests (including the existing ones for `computeExitCode()`) unit tests for `ConsoleLauncherExecutionResult` directly and mock `TestExecutionSummary`.

Reply:
I have created the unit tests for testing `computeExitCode()` in `ConsoleLauncherExecutionResult`. Though I have not removed the ones in `ConsoleTestExecutorTests` despite they are also using `computeExitCode()`. They seem to be conseptually different to me ie they are sort of small 'integration' tests starting from individual tests ie a succeding test a failing tests etc. On the other hand `computeExitCode()` only uses takes into account the `totalFailureCount`.